client/src/main/java/io/confluent/kafka/schemaregistry/client/security/bearerauth/oauth/OauthCredentialProvider.java

```diff
@@ -0,0 +1,117 @@
+ */
+public class OauthCredentialProvider implements BearerAuthCredentialProvider {
+
+  private CachedOauthTokenRetriever tokenRetriever;
+  private String targetSchemaRegistry;
+  private String targetIdentityPoolId;
+
+  @Override
+  public String alias() {
+    return "OAUTHBEARER";
+  }
+
+  @Override
+  public String getBearerToken(URL url) {
+    return tokenRetriever.getToken();
+  }
+
+  @Override
+  public String getTargetSchemaRegistry() {
+    return this.targetSchemaRegistry;
+  }
+
+  @Override
+  public String getTargetIdentityPoolId() {
+    return this.targetIdentityPoolId;
+  }
+
+  @Override
+  public void configure(Map<String, ?> map) {
+    ConfigurationUtils cu = new ConfigurationUtils(map);
+
+    targetSchemaRegistry = cu.validateString(
+        SchemaRegistryClientConfig.BEARER_AUTH_LOGICAL_CLUSTER);
+    targetIdentityPoolId = cu.validateString(
+        SchemaRegistryClientConfig.BEARER_AUTH_IDENTITY_POOL_ID);
+
+    tokenRetriever = new CachedOauthTokenRetriever();
+    tokenRetriever.configure(getTokenRetriever(cu), getTokenValidator(), getOauthTokenCache(map));
+  }
+
+  private OauthTokenCache getOauthTokenCache(Map<String, ?> map) {
+    short cacheExpiryBufferSeconds = SchemaRegistryClientConfig
+        .getBearerAuthCacheExpiryBufferSeconds(map);
+    return new OauthTokenCache(cacheExpiryBufferSeconds);
+  }
+
+  private AccessTokenRetriever getTokenRetriever(ConfigurationUtils cu) {
+
+    String clientId = cu.validateString(SchemaRegistryClientConfig.BEARER_AUTH_CLIENT_ID);
+    String clientSecret = cu.validateString(SchemaRegistryClientConfig.BEARER_AUTH_CLIENT_SECRET);
+    String scope = cu.validateString(SchemaRegistryClientConfig.BEARER_AUTH_SCOPE , false);
+
+    //Keeping following configs needed by HttpAccessTokenRetriever as constants and not exposed to
+    //users for modifications
+    Long retryBackoffMs = SaslConfigs.DEFAULT_SASL_LOGIN_RETRY_BACKOFF_MS;
+    Long retryBackoffMaxMs = SaslConfigs.DEFAULT_SASL_LOGIN_RETRY_BACKOFF_MAX_MS;
+    Integer loginConnectTimeoutMs = null;
+    Integer loginReadTimeoutMs = null;
+    JaasOptionsUtils jou = new JaasOptionsUtils(Collections.EMPTY_MAP);
+    SSLSocketFactory sslSocketFactory = null;
+    URL url = cu.validateUrl(SchemaRegistryClientConfig.BEARER_AUTH_ISSUER_ENDPOINT_URL);
+    if (jou.shouldCreateSSLSocketFactory(url)) {
+      sslSocketFactory = jou.createSSLSocketFactory();
+    }
+
+    return new HttpAccessTokenRetriever(clientId, clientSecret, scope, sslSocketFactory,
+        url.toString(), retryBackoffMs, retryBackoffMaxMs, loginConnectTimeoutMs, loginReadTimeoutMs
+    );
+  }
+
+  private AccessTokenValidator getTokenValidator() {
+    //Keeping following configs needed by LoginAccessTokenValidator as constants and not exposed to
+    //users for modifications
+    String scopeClaimName = SaslConfigs.DEFAULT_SASL_OAUTHBEARER_SCOPE_CLAIM_NAME;
+    String subClaimName = SaslConfigs.DEFAULT_SASL_OAUTHBEARER_SUB_CLAIM_NAME;
```

Comment:
Why we're not exposing these configs for modifications? If the Kafka client support this we should align with it.

Reply:
added it as new schema registry client config